examples/papers/cazalis.jl

```diff
@@ -0,0 +1,91 @@
+# Model of graphene confined to 2 spatial dimensions studied
+# in the paper by Cazalis (arxiv, 2022, TODO add ref)
+# The pure 3D Coulomb 1/|x| interaction is used, without pseudopotential.
+
+using DFTK
+using Plots
+
+## redefine Hartree interaction
+struct Hartree2D end
+struct Term2DHartree <: DFTK.TermNonlinear end
+(t::Hartree2D)(basis) = Term2DHartree()
+function DFTK.ene_ops(term::Term2DHartree, basis::PlaneWaveBasis{T},
+                 ψ, occ; ρ, kwargs...) where {T}
```

Comment:
Because it's not really clear what "2D Hartree" should mean (is it 1/|x| kernel, or is it the solution of Poisson's equation?), so it's better to be explicit here